docker/build-tools/Dockerfile

```diff
@@ -57,30 +59,60 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     xz-utils
 
 # Install protoc
-WORKDIR /tmp
-ADD https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOBUF_VERSION}/protoc-${PROTOBUF_VERSION}-linux-x86_64.zip /tmp/
-RUN unzip /tmp/protoc-${PROTOBUF_VERSION}-linux-x86_64.zip
+ADD https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-linux-x86_64.zip /tmp/
+RUN unzip /tmp/protoc-${PROTOC_VERSION}-linux-x86_64.zip
+RUN mkdir -p ${OUTDIR}/usr/bin
+RUN mv /tmp/bin/protoc ${OUTDIR}/usr/bin
 
 # Install necessary protoc includes
-RUN mkdir -p ${OUTDIR}/usr/include/protobuf/google/protobuf
-RUN mkdir -p ${OUTDIR}/usr/include/protobuf/google/rpc
-RUN mkdir -p ${OUTDIR}/usr/include/protobuf/google/api
-RUN mkdir -p ${OUTDIR}/usr/include/protobuf/gogoproto
-RUN mkdir -p ${OUTDIR}/usr/include/protobuf/validate
+# TODO: Need to pin include file versions
 
+RUN mkdir -p ${OUTDIR}/usr/include/protobuf/google/protobuf
 RUN for f in any duration descriptor empty struct timestamp wrappers; do \
             curl -L -o ${OUTDIR}/usr/include/protobuf/google/protobuf/${f}.proto https://raw.githubusercontent.com/google/protobuf/master/src/google/protobuf/${f}.proto; \
         done
+
+RUN mkdir -p ${OUTDIR}/usr/include/protobuf/google/rpc
 RUN for f in code error_details status; do \
             curl -L -o ${OUTDIR}/usr/include/protobuf/google/rpc/${f}.proto https://raw.githubusercontent.com/istio/gogo-genproto/master/googleapis/google/rpc/${f}.proto; \
         done
+
+RUN mkdir -p ${OUTDIR}/usr/include/protobuf/google/api
 RUN for f in http annotations; do \
             curl -L -o ${OUTDIR}/usr/include/protobuf/google/api/${f}.proto https://raw.githubusercontent.com/istio/gogo-genproto/master/googleapis/google/api/${f}.proto; \
         done
+
+RUN mkdir -p ${OUTDIR}/usr/include/protobuf/validate
 RUN curl -L -o ${OUTDIR}/usr/include/protobuf/validate/validate.proto https://raw.githubusercontent.com/envoyproxy/protoc-gen-validate/${PROTOC_GEN_VALIDATE_VERSION}/validate/validate.proto
+
+RUN mkdir -p ${OUTDIR}/usr/include/protobuf/gogoproto
 RUN curl -L -o ${OUTDIR}/usr/include/protobuf/gogoproto/gogo.proto https://raw.githubusercontent.com/gogo/protobuf/master/gogoproto/gogo.proto
 
-# Install a bunch of Go tools
+RUN mkdir -p ${OUTDIR}/usr/include/protobuf/k8s.io/api
```

Comment:
some gold plating would help here explain what this section does in terms of a comment.